### DISCUSSION TITLE
Sentinel data hub service deprecated

### DISCUSSION THREAD

**Discussion by Xiaohua-Eric-Xu** [gmtsar/gmtsar#852] [Announcements]
ESA has moved all its service to dataspace.copernicus.eu, so any script involves automatically downloading files or orbits will fail before we make another fix. 

So far we haven't been able to find a way to download auxiliary files (e.g., precise orbits) without having to register on the website. Will see how things go as we learn more on their API functions.

A temporary fix is to roll back to use some of the older scripts (e.g., organize_files_tops[_linux],csh) to download orbits from ASF instead. Some of them are here:
https://github.com/gmtsar/gmtsar/blob/6.1/gmtsar/csh/organize_files_tops.csh
https://github.com/gmtsar/gmtsar/blob/6.1/gmtsar/csh/organize_files_tops_linux.csh

If you plan to use the scripts in 6.1, you'll also need to use the older version of create_frame_tops.csh too https://github.com/gmtsar/gmtsar/blob/6.1/gmtsar/csh/create_frame_tops.csh . Or you could manually change, in the two scripts above, when calling create_frame_tops.csh, use a polarization mode (e.g., vv) instead of 1 (at the very end of the line).  

Also, two temporary orbit downloading script are added to the repo: 
https://github.com/gmtsar/gmtsar/blob/master/gmtsar/csh/download_sentinel_orbits_step.csh
https://github.com/gmtsar/gmtsar/blob/master/gmtsar/csh/download_sentinel_orbits_step_linux.csh
(The two scripts above have now been merged to https://github.com/gmtsar/gmtsar/blob/master/gmtsar/csh/download_sentinel_orbits.csh and https://github.com/gmtsar/gmtsar/blob/master/gmtsar/csh/download_sentinel_orbits_linux.csh)

UPDATE: A temporary fix was made to the scripts to enable automatic orbit downloading for Sentinel-1 data again in the master branch (2024-04-14): https://github.com/gmtsar/gmtsar/commit/9d18a750f62a3b9d0b79413a2ca782f1de4e3a0f

**Comment by johntruckenbrodt** [comment]
Hi @Xiaohua-Eric-Xu, do you know this page?
https://step.esa.int/auxdata/orbits/Sentinel-1/

**Reply by Xiaohua-Eric-Xu** [reply]
Yup, the following link are about the same, which is used in earlier scripts.
https://s1qc.asf.alaska.edu/

**Reply by johntruckenbrodt** [reply]
Cool, I didn't know this link yet. Thanks

**Reply by Xiaohua-Eric-Xu** [reply]
I guess yours are better as it does not ask for username and password. :)

**Comment by Xiaohua-Eric-Xu** [comment]
Temporarily added two scripts to download orbits from https://step.esa.int/auxdata/orbits/Sentinel-1. 8df3cc255e116ea3f52181cf4941f1becbf14eac

**Comment by Asmit-iirs** [comment]
Those who are still facing problem and have already setup a [https://dataspace.copernicus.eu/](url) account can refer to the following repo for downloading orbits https://github.com/scottstanie/sentineleof

**Reply by Pavithra-4pi** [reply]
after downloading the precise orbits manually I am not able to run the new organise_files_tops.csh .. has anybody been successful ?

**Reply by Xiaohua-Eric-Xu** [reply]
Are you running that on a Mac or Linux system? For linux you'll need to use organise_files_tops_linux.csh.

**Reply by Pavithra-4pi** [reply]
I am running it on Mac OS Sonoma 14.0

**Reply by Xiaohua-Eric-Xu** [reply]
Did you follow the steps mentioned above? i.e. switch to an older version of the scripts, both organize_files_tops.csh and create_frame_tops.csh. Then alias your asf user and password inside the script.

**Reply by Pavithra-4pi** [reply]
Hi Eric,
I uninstalled gmtsar 6.3 and reinstalled 6.1 version 
when I placed https://github.com/gmtsar/gmtsar/blob/master/gmtsar/csh/download_sentinel_orbits_step.csh in usr/local/bin path the file worked, but later organise organise_files_tops.csh still doesn't work.. I am getting the following error:
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]

**Reply by Xiaohua-Eric-Xu** [reply]
You should be using the linux version of this script, organize_files_tops_linux.csh. (Sorry somehow this message did not show up in my email, maybe I accidentally deleted it somehow)

**Comment by DD-ZHAO** [comment]
@Xiaohua-Eric-Xu 
Dear Professor Xu
I am Ruiqi Zhao from Southwest Jiaotong University. I have previously consulted with you via email about some issues. Due to ESA integrating data into dataspace.coprnicus.eu, an error occurred when downloading precise orbit data. With your help, I have replaced the three scripts with the old version, but now in the data folder to execute organise_files_tops .csh SAFE_filelist ... /reframed/pins.ll 1 command, an orbits.list and tmprecord file were generated in the data folder. The program reported an error and did not successfully download the precision orbit file. Then, execute organise_files_tops .csh SAFE_filelist ... /reframed/pins.ll 2 command still reports an error.
I would like to ask you how to solve this problem. Thank you for your time!

This is the content of the data folder
<img width="543" alt="2" src="https://github.com/gmtsar/gmtsar/assets/111511562/8be3804a-b117-4790-9fbf-10c105a504a4">


This is an error message
<img width="347" alt="1" src="https://github.com/gmtsar/gmtsar/assets/111511562/9f3fe298-676f-407b-af7a-c35d91741859">

**Reply by Xiaohua-Eric-Xu** [reply]
> Are you running that on a Mac or Linux system? For linux you'll need to use organise_files_tops_linux.csh.

See above.

**Reply by DD-ZHAO** [reply]
I'm running it in linux but running organise_files_tops_linux.csh also gives an error

**Reply by Xiaohua-Eric-Xu** [reply]
Which version of the script did you run?

**Reply by DD-ZHAO** [reply]
I ran the script you provided on this page as follows:
https://github.com/gmtsar/gmtsar/blob/6.1/gmtsar/csh/organize_files_tops.csh
https://github.com/gmtsar/gmtsar/blob/6.1/gmtsar/csh/organize_files_tops_linux.csh

**Reply by Xiaohua-Eric-Xu** [reply]
The figure you showed seems to indicate you are using organize_files_tops.csh rather than organize_files_tops_linux.csh, and also it is saying that the `date` options you used does not match the usage as it is written for mac not linux.

**Reply by DD-ZHAO** [reply]
Thank you very much for your reply, I'm double checking the script I'm using

**Comment by xander9797** [comment]
oh，thanks
